db-services/db-diagram-service/src/main/java/com/qreal/wmp/db/diagram/mapping/RelationMapping.java

```diff
@@ -0,0 +1,95 @@
+package com.qreal.wmp.db.diagram.mapping;
+
+import com.qreal.wmp.db.diagram.dao.DiagramDao;
+import com.qreal.wmp.db.diagram.exceptions.NotFoundException;
+import com.qreal.wmp.db.diagram.model.Folder;
+import com.qreal.wmp.thrift.gen.TFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+/** Performs loading of folder's parent for folder converting.*/
+public class RelationMapping {
+
+    private static final Logger logger = LoggerFactory.getLogger(RelationMapping.class);
+
+    private DiagramDao diagramDao;
+
+    public RelationMapping(DiagramDao diagramDao) {
+        this.diagramDao = diagramDao;
+    }
+
+    public Folder convertTFolder(TFolder tFolder) {
+        Folder folder = new Folder(tFolder);
+        folder = loadParents(folder);
+        return folder;
+    }
+
+    /** Loads parent folders for converted from TFolder folder.*/
+    private Folder loadParents(Folder folder) {
+        if (folder.getId() != null) {
+            folder = loadFolderSaved(folder);
+        }
+        if (folder.getFolderParentId() != null) {
+            folder = loadFolderParent(folder);
+        }
+
+        for (Object child : folder.getChildrenFolders().toArray()) {
```

Comment:
this code looks messy with all these casts. may be something like
`folder.setChildrenFolders(folder.getChildrenFolders().stream().map(child -> loadParents(child)).collect(Collectors.toSet()));`
?